client/my-sites/theme/main.jsx

```diff
@@ -1053,9 +1056,10 @@ class ThemeSheet extends Component {
 			);
 		} else if ( isLoggedIn && siteId ) {
 			if (
-				( ! isThemeAllowedOnSite || isPremium ) &&
-				! isThemePurchased &&
-				! isExternallyManagedTheme
+				( ( ! isThemeAllowedOnSite || isPremium ) &&
+					! isThemePurchased &&
+					! isExternallyManagedTheme ) ||
+				( isBundledSoftwareSet && ! canInstallPlugins )
```

Comment:
Hey, @renatho!

This particular change may have unintentionally caused a bug in Woo Express, where themes like `tsubaki` and `tazza` now requires an upgrade.

To reproduce:

1. Create a Woo Express free trial site by going to https://woo.com/start and follow through the NUX process
2. Go to Appearance > Themes
3. Search for `tazza`
4. Select Tazza theme
5. Observe you can't activate it now, despite it being free

<img width="806" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3747241/1ad399e9-7dce-4e7a-bf05-6d794909be78">

Reply:
I've made a PR to fix this, https://github.com/Automattic/wp-calypso/pull/85890.

Appreciate if you can take a look as well, @renatho!

Reply:
@ilyasfoo

Thank you for catching it and quickly raising a PR with the fix!

> May I know what was the intended themes to be matched with this logic?

It was supposed to match the paid themes (previous logic) + bundled themes. But I was assuming that any bundled theme would need to be upgraded to a plan where the user can install plugins because it will install a plugin along with the theme. But I missed this use case that you mentioned.

> I've made a PR to fix this, https://github.com/Automattic/wp-calypso/pull/85890.

I'll take a look now.

Reply:
> May I know what was the intended themes to be matched with this logic?

The new theme we are considering outside of WooCommerce is the Course theme, which installs the Sensei LMS.